pkg/operator/events/recorder.go

```diff
@@ -40,13 +41,24 @@ func GetControllerReferenceForCurrentPod(client corev1client.PodInterface) (*cor
 	if err != nil {
 		return nil, err
 	}
-	ownerRef := metav1.GetControllerOf(pod)
+	replicaSetOwnerRef := metav1.GetControllerOf(pod)
+	if replicaSetOwnerRef == nil || replicaSetOwnerRef.Kind != "ReplicaSet" {
+		return nil, fmt.Errorf("pod %s is not controlled by replicaSet", podName)
```

Comment:
this isn't an error, just return what you have.  We have operators that daemonsets for instance

Reply:
@deads2k oh, I forgot about daemonsets... will need to handle that case as well